### PR TITLE
Justin2 Pull Request

### PR DESCRIPTION
**Pull request by 9Mcgusto6**
Finished the two BorrowBook files. Need to have them reviewed for the final scheduled merge.

### CODE REVIEW

File: src/library/borrowbook/BorrowBookUI.java [hunk 1]
```diff
@@ -4,115 +4,115 @@
...
-					CoNtRoL.CaNcEl();
-					
-				} else {
-					CoNtRoL.CoMmIt_LoAnS();
-					iNpUT("Press <any key> to complete ");
-				}
-				break;
+            case FINALISING:
+                String ans = input("Commit loans? (Y/N): ");
+                if (ans.toUpperCase().equals("N")) {
```
Comment (michaelbarrell) [inline src/library/borrowbook/BorrowBookUI.java]: ans isn't really a descriptive variable name

File: src/library/payfine/PayFineControl.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ public double payFine(double amount) {
             UI.displayOutput(String.format("Change: $%.2f", change));
         }
         UI.displayOutput(member.toString());
-        UI.setState(PayFineUI.UI_STATE.COMPLETED);
+        UI.setState(PayFineUI.UIState.COMPLETED);
         state = ControlState.COMPLETED;
         return change;
     }
```
Comment (michaelbarrell) [inline src/library/payfine/PayFineControl.java:58]: Thank you for fixing my mistakes

**Comment by Firedingo** [comment]
As mediator, I confirm that the code runs and the review can proceed

File: src/library/borrowbook/BorrowBookUI.java [hunk 3]
```diff
@@ -4,115 +4,115 @@
...
+                String memberString = input("Swipe member card (press <enter> to cancel): ");
+                if (memberString.length() == 0) {
+                    control.cancelOperation();
+                    break;
+                }
+                try {
+                    int memberId = Integer.valueOf(memberString).intValue();
+                    control.cardSwiped(memberId);
+                }
+                catch (NumberFormatException exit) {
```
Comment (michaelbarrell) [inline src/library/borrowbook/BorrowBookUI.java]: consider changing exit to exception

File: src/library/borrowbook/BorrowBookUI.java [hunk 4]
```diff
@@ -4,115 +4,115 @@
...
+                String bookStringInput = input("Scan Book (<enter> completes): ");
+                if (bookStringInput.length() == 0) {
+                    control.completeOperation();
+                    break;
+                }
+                try {
+                    int bookId = Integer.valueOf(bookStringInput).intValue();
+                    control.bookScanned(bookId);					
+                } 
+                catch (NumberFormatException exit) {
```
Comment (michaelbarrell) [inline src/library/borrowbook/BorrowBookUI.java]: consider changing exit to exception

File: src/library/borrowbook/BorrowBookUI.java (hunk 1 above)
Comment (michaelbarrell) [reply src/library/borrowbook/BorrowBookUI.java]: consider changing to answer

**Review by michaelbarrell — COMMENTED** [review]
As discussed in project chat. Files to be changed to conform to code style guidelines:
Main.java: 	
	change sb to stringBuilder lines 30, 32, and 51
	change mm to MM line 60
	change e to exception lines 193 and 219 

ReturnBookUI.java:
	change e to exception line 42
	change ans to answer lines 49 and 51

BorrowBookUI.java
	change exit to exception lines 60 and 82
	change ans to answer lines 89 and 90

Book.java
	remove whitespace from enum braces line 13
	change sb to stringBuilder lines 26, 27, and 33

Calendar.java
	change e to exception lines 33, 34, 46, and 47

Library.java
	change e to exception lines 62, 63, 82, and 83
	make variable names meaningful lines 135, 136, 137, and 138
	add curly braces to else statement line 245

Loan.java
	change sdf to simpleDateFormat lines 48 and 56
	change sb to stringBuilder lines 50, 51, and 58
Awaiting mediator response.

File: src/library/borrowbook/BorrowBookUI.java (hunk 3 above)
Comment (Firedingo) [reply src/library/borrowbook/BorrowBookUI.java]: Resolution: https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/38d55bba36e203d850141f49122bcc9d8d7af203

File: src/library/borrowbook/BorrowBookUI.java (hunk 4 above)
Comment (Firedingo) [reply src/library/borrowbook/BorrowBookUI.java]: Resolution: https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/38d55bba36e203d850141f49122bcc9d8d7af203

File: src/library/borrowbook/BorrowBookUI.java (hunk 1 above)
Comment (Firedingo) [reply src/library/borrowbook/BorrowBookUI.java]: Resolution: https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/3e46915fa39c52dda82e4a072dec5c38df902734

**Comment by Firedingo** [comment]
> As discussed in project chat. Files to be changed to conform to code style guidelines:
> Main.java:
> change sb to stringBuilder lines 30, 32, and 51
> change mm to MM line 60
> change e to exception lines 193 and 219
> 
> ReturnBookUI.java:
> change e to exception line 42
> change ans to answer lines 49 and 51
> 
> BorrowBookUI.java
> change exit to exception lines 60 and 82
> change ans to answer lines 89 and 90
> 
> Book.java
> remove whitespace from enum braces line 13
> change sb to stringBuilder lines 26, 27, and 33
> 
> Calendar.java
> change e to exception lines 33, 34, 46, and 47
> 
> Library.java
> change e to exception lines 62, 63, 82, and 83
> make variable names meaningful lines 135, 136, 137, and 138
> add curly braces to else statement line 245
> 
> Loan.java
> change sdf to simpleDateFormat lines 48 and 56
> change sb to stringBuilder lines 50, 51, and 58
> Awaiting mediator response.

Resolutions:

Main.java
- Defect #1 -  https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/62567551b261dc9fe974b5d91741ec33e7f2458c
- Defect #2 - https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/610b3a1091038e2e31068f56130abbffc6a89352
- Defect #3 - https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/2b4f6b8d020b5140846dfcb7106c6cbc82286188

ReturnBookUI.java
- Defect #4 - https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/870d12e2d6eae4b64a06fa7bff24712aeae7f924
- Defect #5 - https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/98d261d2c426b9b2fe06db708a3ad126290fc664

Book.java
- Defect #6 - https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/b825beec459f7428593aa62bccbb81c98145d1a4
- Defect #7 - https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/b825beec459f7428593aa62bccbb81c98145d1a4

Calendar.java
- Defect #8 - https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/50a3c8d52054d156a66b608941aae1562a577f68

Library.java
- Defect #9 - https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/2718f0962239b2db5e96cdd1c68124701af624fb
- Defect #10 - https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/62a452da9d8aa37889992df07d107d3bd245911b
- Defect #11 - https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/efa8228c55227aa274cc07424bc8da1e044ae18d

Loan.java
- Defect #12 - https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/57f2a74c7f20935d3c6775574e45f6c6836be216
- Defect #13 - https://github.com/9Mcgusto6/ITC205-Team17/pull/6/commits/6385254efd1e5a95f88bcb8b7ac900d5ee261b32

**Comment by Firedingo** [comment]
Looks Good to me as mediator. Code runs with the fixes, I see no reason for the merge not to go ahead.

**Comment by Firedingo** [comment]
Master still works post merge also!

**Comment by Firedingo** [comment]
Master still works post merge also!

**Comment by Firedingo** [comment]
Master still works post merge also!

**Pull request merged**